core/src/main/scala/com/databricks/labs/remorph/parsers/tsql/TSqlAstBuilder.scala

```diff
@@ -29,10 +29,8 @@ class TSqlAstBuilder extends TSqlParserBaseVisitor[ir.TreeNode] {
       .select_list_elem()
       .asScala
       .map(_.accept(new TSqlExpressionBuilder))
-    val fromTable = rawExpression.table_sources().accept(new TSqlRelationBuilder)
-
-    ir.Project(fromTable, columnExpression)
-
+    val tableSources = Option(rawExpression.table_sources())
+      .fold(ir.NoTable().asInstanceOf[ir.Relation])(_.accept(new TSqlRelationBuilder))
```

Comment:
```suggestion
      .fold[ir.Relation](ir.NoTable())(_.accept(new TSqlRelationBuilder))
```

Reply:
Type inference indeed needs a little help here, but there's always a way to avoid asInstanceOf ;)